lib/src/Uncertainty/Algorithm/MetaModel/openturns/MetaModelAlgorithm.hxx

```diff
@@ -47,12 +47,21 @@ public:
   MetaModelAlgorithm();
 
   /** Constructor with parameters */
+  MetaModelAlgorithm(const Sample & inputSample,
+                     const Sample & outputSample);
+
+  MetaModelAlgorithm(const Sample & inputSample,
+                     const Point & weights,
+                     const Sample & outputSample);
+
   MetaModelAlgorithm(const Sample & inputSample,
                      const Sample & outputSample,
                      const Distribution & distribution);
 
   MetaModelAlgorithm(const Sample & inputSample,
-                     const Sample & outputSample);
+                     const Point & weights,
+                     const Sample & outputSample,
+                     const Distribution & distribution);
```

Comment:
would it be feasible to pass the weights via a separate accessor, instead of multiplying ctors ?

Reply:
When using a WeightedExperiment, the points and the weights come together, so I prefer to allow to set them together in the ctor